lib/MainTemplate.js

```diff
@@ -399,8 +399,14 @@ module.exports = class MainTemplate extends Tapable {
 		this.hooks.hashForChunk.call(hash, chunk);
 	}
 
+	// TODO remove in webpack 5
 	useChunkHash(chunk) {
 		const paths = this.hooks.globalHashPaths.call([]);
 		return !this.hooks.globalHash.call(chunk, paths);
 	}
+
+	useGlobalHash(chunk) {
+		const paths = this.hooks.globalHashPaths.call([]);
+		return this.hooks.globalHash.call(chunk, paths);
```

Comment:
Could you check for `undefined` instead of relying on the truthiness of the result?